profiling/src/profiling/stalk_walking.rs

```diff
@@ -123,3 +123,33 @@ pub(super) unsafe fn collect_stack_sample(
     }
     Ok(samples)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::bindings as zend;
+
+    #[test]
+    #[cfg(feature = "stack_walking_tests")]
+    fn test_collect_stack_sample() {
+        unsafe {
+            let fake_execute_data = zend::ddog_php_test_create_fake_zend_execute_data(3);
```

Comment:
This is super nice! 
This will also allow us to have better tests around what we can expect from zend_execute_data.